### PR TITLE
LPS-79130

### PR DESCRIPTION
**Pull request by ricardocousolr**
Hi @pei-jung. This fix is for a customer. 

The idea is to reindex all the users within an organization or a user group that has become or stops being a member of site, so that they show in the search results. 

Two things to notices about the solution: 

1. It doesn't cover the case of organizations within organizations. Since this isn't the customer's case, we could deal with it in another ticket. 
2. If the number of users in a user group or oganization is large, reindexing could be a costly operation and it might be best to do it as a background task. We could also deal with this in a another ticket, as an improvement. 

Tell me if you have any questions. 
Thanks!

### CODE REVIEW

**Comment by ricardocousolr** [comment]
ci:test

**Comment by pei-jung** [comment]
ci:test

**Comment by pei-jung** [comment]
ci:reopen

**Comment by pei-jung** [comment]
ci:retest

**Comment by ricardocousolr** [comment]
ci:retest

**Comment by pei-jung** [comment]
@jorgeferrer @drewbrokke Can you take a look at this? I'd like your opinion on this as well. Do you think we should index all the groups that a user belongs to, including the ones implicitly associated via organizations or user groups? As Ricardo mentioned in his comment, this could be a costly operation if the number of users in the organizaiton/user group is huge.

File: portal-impl/src/com/liferay/portal/service/impl/GroupLocalServiceImpl.java [hunk 1]
```diff
@@ -1114,6 +1114,11 @@ public Group fetchUserPersonalSiteGroup(long companyId)
 		return groupPersistence.findByC_A(companyId, active);
 	}
 
+	@Override
+	public List<Long> getActiveTransitiveGroupIds(long userId) {
+		return groupFinder.findByActiveTransitiveGroupIds(userId);
```
[the remark this answers is not in the document]
Comment (ricardocousolr) [reply portal-impl/src/com/liferay/portal/service/impl/GroupLocalServiceImpl.java]: You're right, using `GroupLocalServiceUtil.getUserGroups(long userId, boolean inherit)` with `inherit = true` (I guess you meant `true` and not `false`) gives the same information and actually more: I wasn't taking into account the case of organization groups. 

So I have redone the pr and is ready to send but if you like we can wait on the feedback about the indexing cost.

**Comment by JorgeFerrer** [comment]
I am concerned with the scalability implications of this solution. I believe we should send it to the Core Infra team to run the performance testing over it before sending it to master.

File: portal-impl/src/com/liferay/portal/service/impl/GroupLocalServiceImpl.java [hunk 2]
```diff
@@ -508,6 +508,150 @@ public Group addGroup(
...
 
+	/**
+	 * Adds the organization to the group.
+	 *
+	 * @param organizationId the primary key of the organization
+	 * @param group the primary key of the group
+	 */
+	@Override
+	public void addOrganizationGroup(long organizationId, Group group) {
+		organizationPersistence.addGroup(organizationId, group);
```
Comment (drewbrokke) [inline portal-impl/src/com/liferay/portal/service/impl/GroupLocalServiceImpl.java]: I think we should be calling `super` in all of these.

**Comment by pei-jung** [comment]
Hi @ricardocousolr,

Can you take a look at the comments left by Drew and make the changes accordingly? After you're done, please send a pull request to @shuyangzhou's team for performance testing as suggested by Jorge. Let me know if you have any questions. Thanks!

**Comment by ricardocousolr** [comment]
Thanks for all the comments. I'll integrate them in the new pr and send it along.

**Comment by ricardocousolr** [comment]
Sent to https://github.com/shuyangzhou/liferay-portal/pull/5999

**Comment by shuyangzhou** [comment]
See https://github.com/brianchandotcom/liferay-portal/compare/master...shuyangzhou:LPS-79130

**Pull request closed without merging**